### ISSUE TITLE
Python 2.7.9 breaks ssl_wrap_socket

### ISSUE THREAD

**Issue by Lukasa** [shazow/urllib3#543]
The following code is in `urllib3/util/ssl_.py`, in `ssl_wrap_socket`:

```python
if HAS_SNI:  # Platform-specific: OpenSSL with enabled SNI
    return context.wrap_socket(sock, server_hostname=server_hostname)
return context.wrap_socket(sock)
```

In essence, we only pass the server hostname if SNI is available on the platform.

Unfortunately, this breaks on Python 2.7.9, as you can see in this requests traceback:

```
Python 2.7.9 (default, Feb  2 2015, 17:03:23)
[GCC 4.1.2 20080704 (Red Hat 4.1.2-52)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> r = requests.get('https://api.github.com/user', auth=('user', 'pass'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/api.py", line 65, in get
    return request('get', url, **kwargs)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/api.py", line 49, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/adapters.py", line 370, in send
    timeout=timeout
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/packages/urllib3/connectionpool.py", line 518, in urlopen
    body=body, headers=headers)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/packages/urllib3/connectionpool.py", line 322, in _make_request
    self._validate_conn(conn)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/packages/urllib3/connectionpool.py", line 727, in _validate_conn
    conn.connect()
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/packages/urllib3/connection.py", line 238, in connect
    ssl_version=resolved_ssl_version)
  File "/opt/python2.7/lib/python2.7/site-packages/requests-2.5.1-py2.7.egg/requests/packages/urllib3/util/ssl_.py", line 254, in ssl_wrap_socket
    return context.wrap_socket(sock)
  File "/opt/python2.7/lib/python2.7/ssl.py", line 350, in wrap_socket
    _context=self)
  File "/opt/python2.7/lib/python2.7/ssl.py", line 537, in __init__
    raise ValueError("check_hostname requires server_hostname")
ValueError: check_hostname requires server_hostname
```

It seems that with `check_hostname` set to true (the default on Python 2.7.9) we _must_ pass the `server_hostname` argument.

Not really sure what the best fix is here.

Follow-on from kennethreitz/requests#2435 and @mjpieters in #517.

**Comment by t-8ch** [comment]
This should also be fixed by #526 or something equivalent

**Comment by Lukasa** [comment]
Agreed, #526 should fix it.
